salt/states/saltmod.py

```diff
@@ -663,26 +663,50 @@ def wait_for_event(name, id_list, event_id="id", timeout=300, node="master"):
                     val = event["data"]["data"].get(event_id)
 
                 if val is not None:
-                    try:
-                        val_idx = id_list.index(val)
-                    except ValueError:
-                        log.trace(
-                            "wait_for_event: Event identifier '%s' not in "
-                            "id_list; skipping.",
-                            event_id,
-                        )
+                    if isinstance(val, list):
+
+                        val_list = [id for id in id_list if id in val]
+
+                        if len(val_list) == 0:
+                            log.trace(
+                                "wait_for_event: Event identifier '%s' not in "
+                                "id_list; skipping",
+                                event_id,
+                            )
+                        elif len(val_list) > 0:
```

Comment:
```suggestion
                        elif val_list:
```

Reply:
```suggestion
                        else:
```
This might even be more appropriate.